packages/miniflare/test/plugins/kv/assets.spec.ts

```diff
@@ -0,0 +1,220 @@
+		}
+
+		const mf = new Miniflare({
+			...options,
+			scriptPath: t.context.modulesPath,
+			modules: true,
+			assetsPath: tmp,
+		});
+		t.teardown(() => mf.dispose());
+
+		for (const [route, expectedContents] of Object.entries(routeContents)) {
+			const res = await mf.dispatchFetch(`http://localhost:8787${route}`);
+			const expected = expectedRoutes.has(route as Route);
+			const text = (await res.text()).trim();
+			t.is(res.status, expected ? 200 : 404, `${route}: ${text}`);
+			if (expected) t.is(text, expectedContents, route);
+		}
+	},
+};
+
+test(
+	"gets all assets",
+	getMacro,
+	{},
+	new Set<Route>(["/", "/a.txt", "/b/b.txt"])
+);
+
+// Tests for checking different types of globs are matched correctly
+const matchMacro: Macro<[string], Context> = {
+	async exec(t, include) {
+		const tmp = await useTmp(t);
+		const dir = path.join(tmp, "a", "b", "c");
+		await fs.mkdir(dir, { recursive: true });
+		await fs.writeFile(path.join(dir, "test.txt"), "test", "utf8");
+		const mf = new Miniflare({
+			siteInclude: [include],
+			scriptPath: t.context.modulesPath,
+			modules: true,
+			assetsPath: tmp,
+		});
+		t.teardown(() => mf.dispose());
+		const res = await mf.dispatchFetch("http://localhost:8787/a/b/c/test.txt");
+		t.is(res.status, 200);
+		await res.arrayBuffer();
+	},
+};
+
+test("matches file name pattern", matchMacro, "test.txt");
+test("matches exact pattern", matchMacro, "a/b/c/test.txt");
+test("matches extension patterns", matchMacro, "*.txt");
+test("matches globstar patterns", matchMacro, "**/*.txt");
+test("matches wildcard directory patterns", matchMacro, "a/*/c/*.txt");
+
+test("doesn't cache assets", async (t) => {
+	const tmp = await useTmp(t);
+	const testPath = path.join(tmp, "test.txt");
+	await fs.writeFile(testPath, "1", "utf8");
+
+	const mf = new Miniflare({
+		scriptPath: t.context.modulesPath,
+		modules: true,
+		assetsPath: tmp,
+	});
+	t.teardown(() => mf.dispose());
+
+	const res1 = await mf.dispatchFetch("http://localhost:8787/test.txt");
+	t.is(res1.headers.get("CF-Cache-Status"), "MISS");
+	t.is(await res1.text(), "1");
+
+	await fs.writeFile(testPath, "2", "utf8");
+	const res2 = await mf.dispatchFetch("http://localhost:8787/test.txt");
+	t.is(res2.headers.get("CF-Cache-Status"), "MISS");
+	t.is(await res2.text(), "2");
+});
+
+test("gets assets with percent-encoded paths", async (t) => {
+	// https://github.com/cloudflare/miniflare/issues/326
+	const tmp = await useTmp(t);
+	const testPath = path.join(tmp, "ń.txt");
+	await fs.writeFile(testPath, "test", "utf8");
+	const mf = new Miniflare({
+		scriptPath: t.context.modulesPath,
+		modules: true,
+		assetsPath: tmp,
+	});
+	t.teardown(() => mf.dispose());
+	const res = await mf.dispatchFetch("http://localhost:8787/ń.txt");
+	t.is(await res.text(), "test");
+});
+
+const isWindows = process.platform === "win32";
+const unixTest = isWindows ? test.skip : test;
+unixTest("static content namespace supports listing keys", async (t) => {
```

Comment:
I don't think listing keys is relevant for Workers + Assets, therefore removing...